### ISSUE TITLE
GNU License limitations

### ISSUE THREAD

**Issue by graymatt** [PHPAuth/PHPAuth#126]
Hello,

Your PHP Auth library is awesome - and I would love to use it in one of my client web application projects. But my understanding of the GNU license, is that if I include your PHPAuth, then I also need to release the code to my web app. Which I don't want to do. Would you consider a more open license, like the MIT license?
Thanks!

**Comment by cuonic** [comment]
Good point, I never thought that the GPLv3 license would cause a problem, but in your case it appears that it does, meaning that anyone can legally request the source code of your application on the basis that the application itself contains GPL code (Source: http://programmers.stackexchange.com/a/47048)

I'll take a look at the MIT license and see what I can do.

**Comment by bugada** [comment]
I think this is a pure php Library, if you run this Library in your server then the internal use doesn't oblige you to redistribute the code even with a gpl3 license. If you use this Library for a Project that include it and is redistribute then the whole project must be gpl and the source code made available.

Well i don't think that a gpl3 causes too much legal problems if you use phpauth in your closed webapp, but I think a MIT license would be better

**Comment by cuonic** [comment]
Ok I've had a quick look at the MIT license and it looks more permissive than the GPLv3 when it comes to reusing licensed software in private projects, you do no have to keep the same license and release the source code.

Here's an explanation in plain english: https://tldrlegal.com/license/mit-license

I'll leave this here until Friday, if anyone is against changing over to MIT then go ahead, if no-one is against we'll move over.

**Comment by cuonic** [comment]
No-one is against. Moving over to MIT license.

**Comment by graymatt** [comment]
Thank you so much for taking my concerns in to view, and for updating the license. I think this will really help adoption of PHPAuth. It is much appreciated!

**Comment by cuonic** [comment]
No problem, thanks for contributing !
